core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -392,6 +392,16 @@ static HttpResponse of(Publisher<? extends HttpObject> publisher) {
         }
     }
 
+    /**
+     * Creates a new HTTP response of the specified headers whose content is produced from an existing
+     * {@link Publisher}.
+     */
+    static HttpResponse of(ResponseHeaders headers, Publisher<? extends HttpData> contentPublisher) {
```

Comment:
Don't we need to accept a trailer here?
Which is better:
```java
// May produce a `trailers` at the end of streams
// We need to check that an HttpObject is an instance of HttpData before receiving a `trailers`. 
HttpResponse of(ResponseHeaders headers, Publisher<? extends HttpObject> contentPublisher) {
}

// Or separately take `tailers`?
// This is type-safe but the signature is a bit verbose.
HttpResponse of(ResponseHeaders headers, Publisher<? extends HttpData> contentPublisher, CompletableFuture<HttpHeaders> trailers) {
}
```

Reply:
Because we already have `HttpResponse.of(Publisher<? extends HttpObject>)`, I'd say `HttpResponse.of(ResponseHeaders, Publisher<? extends HttpObject>)` seems better, it's like syntactic sugar for the former (which I think is the purpose of this addition). For the order of data/trailers, isn't it handled in `HttpResponseSubscriber`?

Reply:
>  I'd say HttpResponse.of(ResponseHeaders, Publisher<? extends HttpObject>) seems better

+1

>  For the order of data/trailers, isn't it handled in HttpResponseSubscriber?

Yes. It is already checked by `HttpResponseSubscriber`. `PrependingPublisher` does not need to handle it.